Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelBlueprintCreater.hpp

```diff
@@ -0,0 +1,163 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/Blueprint.hpp"
+#include "Acts/Detector/KdtSurfacesProvider.hpp"
+#include "Acts/Detector/interface/IInternalStructureBuilder.hpp"
+#include "Acts/Geometry/Extent.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/VolumeBounds.hpp"
+#include "Acts/Utilities/BinningData.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace Acts {
+
+struct GeoModelTree;
+
+/// @brief This is the GeoModel blueprint creator for the Gen2 ("Acts::Experimental::Detector")
+/// geometry concept.
+///
+class GeoModelBlueprintCreater {
+ public:
+  /// The configuration struct
+  struct Config {
+    /// The detector surfaces - leave empty if filling is not done
+    /// with a kdtree sorting structure
+    std::vector<std::shared_ptr<Surface>> detectorSurfaces = {};
+    /// The binning values - see above
+    std::vector<BinningValue> kdtBinning = {};
+    /// Polyhedron approximations
+    unsigned int nSegments = 1u;
+  };
+
+  /// The cache struct
+  struct Cache {
+    /// The kdtree of surfaces
+    std::shared_ptr<Experimental::KdtSurfaces<3u>> kdtSurfaces = nullptr;
+  };
+
+  /// The Options struct
+  struct Options {
+    /// The table name of the blueprint in the database
+    std::string table;
+    /// The top level node name
+    std::string topEntry;
+    /// Optionally override the top node bounds
+    std::string topBoundsOverride = "";
+  };
+
+  /// The Blueprint return object
+  struct Blueprint {
+    std::string name;
+    std::unique_ptr<Acts::Experimental::Blueprint::Node> topNode;
+
+    /// Access to the top node
+    const Acts::Experimental::Blueprint::Node& node() const {
+      if (topNode == nullptr) {
+        throw std::runtime_error(
+            "GeoModelBlueprintCreater::Blueprint: No top node created");
+      }
+      return *(topNode.get());
+    }
+  };
+
+  // Blueprint table is:
```

Comment:
This is not really a doxygen comment, Should this go to the *.cpp file? Or should this become a doxygen comment?

If this is only a type used in private functions, it could be also only be forward declared here in the *.hpp and then defined in the *.cpp

Reply:
Turned it into a comment to make it clear what the thing does.